src/video/x11/SDL_x11video.c

```diff
@@ -460,6 +460,14 @@ X11_VideoInit(_THIS)
     X11_InitXfixes(_this);
 #endif /* SDL_VIDEO_DRIVER_X11_XFIXES */
 
+#ifndef X_HAVE_UTF8_STRING
+    SDL_LogError(SDL_LOG_CATEGORY_VIDEO, "X server doesn't support UTF8_STRING, a feature introduced in 2000! This is likely to become a hard error in a future libSDL2.");
```

Comment:
We shouldn't add an error message based on a preprocessor define. Instead this should be a build #warning